src/java/org/apache/cassandra/db/compaction/CompactionTask.java

```diff
@@ -428,4 +403,115 @@ public static long getMaxDataAge(Collection<SSTableReader> sstables)
         }
         return max;
     }
+
+    private void debugLogCompactionSummaryInfo(
+    UUID taskId,
+    long start,
+    long totalKeysWritten,
+    long inputSizeBytes,
+    Collection<SSTableReader> newSStables,
+    long[] mergedRowsHistogram,
+    int totalSourceCQLRows,
+    long outputSizeBytes)
+    {
+        // log a bunch of statistics about the result and save to system table compaction_history
+        long durationInNano = System.nanoTime() - start;
+        long dTime = TimeUnit.NANOSECONDS.toMillis(durationInNano);
+        double ratio = (double) outputSizeBytes / (double) inputSizeBytes;
+
+        long totalSourceRows = 0;
+        StringBuilder mergeSummary = new StringBuilder(mergedRowsHistogram.length * 10);
+        mergeSummary.append('{');
+        for (int i = 0; i < mergedRowsHistogram.length; i++)
+        {
+            long mergedRowCount = mergedRowsHistogram[i];
+            if (mergedRowCount != 0)
+            {
+                totalSourceRows += mergedRowCount * (i + 1);
+                mergeSummary.append(i).append(':').append(mergedRowCount).append(", ");
+            }
+        }
+        mergeSummary.append('}');
+
+        StringBuilder newSSTableNames = new StringBuilder(newSStables.size() * 100);
+        for (SSTableReader reader : newSStables)
+            newSSTableNames.append(reader.descriptor.baseFilename()).append(",");
+        logger.debug(
+        "Compacted ({}) {} sstables to [{}] to level={}." +
+        " {} to {} (~{}% of original) in {}ms." +
+        " Read Throughput = {}, Write Throughput = {}, Row Throughput = ~{}/s." +
+        " {} total partitions merged to {}." +
+        " Partition merge counts were {}",
+        taskId,
+        transaction.originals().size(),
+        newSSTableNames.toString(),
+        getLevel(),
+        prettyPrintMemory(inputSizeBytes),
+        prettyPrintMemory(outputSizeBytes),
+        (int) (ratio * 100),
+        dTime,
+        prettyPrintMemoryPerSecond(inputSizeBytes, durationInNano),
+        prettyPrintMemoryPerSecond(outputSizeBytes, durationInNano),
+        totalSourceCQLRows / (TimeUnit.NANOSECONDS.toSeconds(durationInNano) + 1),
+        totalSourceRows,
+        totalKeysWritten,
+        mergeSummary.toString());
+    }
+
+    private void debugLogCompactingMessage(UUID taskId)
+    {
+        Set<SSTableReader> originals = transaction.originals();
+        StringBuilder ssTableLoggerMsg = new StringBuilder(originals.size() * 100);
+        ssTableLoggerMsg.append("Compacting (").append(taskId).append(')').append(" [");
+        for (SSTableReader sstr : originals)
+        {
+            ssTableLoggerMsg
+            .append(sstr.getFilename())
```

Comment:
This makes better sense on the previous line.

Reply:
Fixed in 142f9b47a8e7fa4c142da836b23df2ae66cad379.